File: app/src/main/java/br/com/levimendesestudos/avenuecode/adapters/CustomInfoWindowAdapter.java
package br.com.levimendesestudos.avenuecode.adapters;

import android.app.Activity;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.model.Marker;
import br.com.levimendesestudos.avenuecode.R;

/**
 * Created by 809778 on 03/10/2016.
 */

public class CustomInfoWindowAdapter implements GoogleMap.InfoWindowAdapter {

    private final View mWindow;
    private final View mContents;

    public CustomInfoWindowAdapter(Activity activity) {
        mWindow   = activity.getLayoutInflater().inflate(R.layout.custom_info_window, null);
        mContents = activity.getLayoutInflater().inflate(R.layout.custom_info_contents, null);
    }

    @Override
    public View getInfoWindow(Marker marker) {
        render(marker, mWindow);
        return mWindow;
    }

    @Override
    public View getInfoContents(Marker marker) {
        render(marker, mContents);
        return mContents;
    }

    private void render(Marker marker, View view) {
        ((ImageView) view.findViewById(R.id.badge)).setImageResource(R.drawable.google_maps_icon);
        String title = marker.getTitle();
        TextView snippetUi = ((TextView) view.findViewById(R.id.snippet));
        snippetUi.setText(title);
    }
}

File: app/src/main/java/br/com/levimendesestudos/avenuecode/adapters/CustomInfoWindowAdapter.kt
package br.com.levimendesestudos.avenuecode.adapters

import android.app.Activity
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.Marker
import br.com.levimendesestudos.avenuecode.R

/**
 * Created by 809778 on 03/10/2016.
 */

class CustomInfoWindowAdapter(var activity: Activity) : GoogleMap.InfoWindowAdapter {


    override fun getInfoWindow(marker: Marker): View {
        val window = activity.layoutInflater.inflate(R.layout.custom_info_window, null)
        render(marker, window)
        return window
    }

    override fun getInfoContents(marker: Marker): View {
        val contents = activity.layoutInflater.inflate(R.layout.custom_info_contents, null)

        render(marker, contents)
        return contents
    }

    private fun render(marker: Marker, view: View) {
        (view.findViewById<View>(R.id.badge) as ImageView).setImageResource(R.drawable.google_maps_icon)
        val title = marker.title
        val snippetUi = view.findViewById<View>(R.id.snippet) as TextView
        snippetUi.text = title
    }
}